bsk_rl/envs/GeneralAEOS/gym_env.py

```diff
@@ -0,0 +1,348 @@
+
+        terminated = False
+        for satellite in self.satellites:
+            if not satellite.is_alive():
+                terminated = True
+                reward += self.failure_penalty
+
+        truncated = False
+        if self.simulator.sim_time >= self.time_limit:
+            truncated = True
+            if self.terminate_on_time_limit:
+                terminated = True
+
+        observation = self._get_obs()
+        info = self._get_info()
+        return observation, reward, terminated, truncated, info
+
+    def render(self) -> None:
+        """No rendering implemented"""
+        return None
+
+    def close(self) -> None:
+        """Try to cleanly delete everything"""
+        if self.simulator is not None:
+            del self.simulator
+
+
+class SingleSatAEOS(GeneralAEOS):
+    """Single satellite action and observation space. Actions and observations are not wrapped in a tuple type."""
+
+    def __init__(self, *args, **kwargs) -> None:
+        super().__init__(*args, **kwargs)
+        assert len(self.satellites) == 1
+
+    @property
+    def action_space(self) -> spaces.Discrete:
+        return self.satellite.action_space
+
+    @property
+    def observation_space(self) -> spaces.Box:
+        super().observation_space
+        return self.satellite.observation_space
+
+    @property
+    def satellite(self) -> Satellite:
+        return self.satellites[0]
+
+    def step(self, action) -> tuple[Any, float, bool, bool, dict[str, Any]]:
+        return super().step([action])
+
+    def _get_obs(self) -> Any:
+        return self.satellite.get_obs()
+
+
+if __name__ == "__main__":
+    import gymnasium as gym
+
+    from bsk_rl.envs.GeneralAEOS.scenario import communication, data
+    from bsk_rl.envs.GeneralAEOS.scenario import satellites as sats
+    from bsk_rl.envs.GeneralAEOS.scenario.environment_features import CityTargets
+    from bsk_rl.envs.GeneralAEOS.simulation import environment
+    from bsk_rl.utilities.initial_conditions import leo_orbit
+
+    env_features = CityTargets(n_targets=5000, location_offset=10e3)
+    data_manager = data.UniqueImagingManager(env_features)
+    oes = leo_orbit.walker_delta(
+        n_spacecraft=3,
+        n_planes=1,
+        rel_phasing=0,
+        altitude=500 * 1e3,
+        inc=45,
+        clustersize=3,
+        clusterspacing=30,
+    )
+    satellites = []
+    sat_type = sats.FullFeaturedSatellite
+    for i, oe in enumerate(oes):
+        sat_args = sat_type.default_sat_args(
+            oe=oe,
+            imageAttErrorRequirement=0.01,
+            imageRateErrorRequirement=0.01,
+        )
+        satellites.append(
+            sat_type("EO" + str(i + 1), sat_args, n_ahead_observe=30, n_ahead_act=15)
+        )
+    # communicator = communication.FreeCommunication(satellites)
+    communicator = communication.LOSMultiCommunication(satellites)
+
+    # Instantiate the environment
+    env = gym.make(
+        "generalAEOS-v1",
```

Comment:
Does this __name__==__main__ work? I can't check it because all of my machines are < Python 3.9 (except for CURC, which isn't the best for debugging anything).

Reply:
Pulled the branch and tested... Python 3.10 annotations do break the other environments. Should discuss a workaround.